marklogic-data-hub-central/ui/src/components/queries/queries.tsx

```diff
@@ -222,6 +224,7 @@ const Query: React.FC<Props> = (props) => {
     }
 
     const onNoClick  = () => {
+        toggleEntityConfirmation(false);
```

Comment:
@xnikhil08 @brucean52 @Sanjeevani19 @timur-isangulov  I think we should close the modal first and then apply current query on entity change.
Earlier the modal was closed after setting the query, hence modal was hiding the spinner and hub properties was getting clicked on prematurely. So the fix was required in the code than to the test. This fixed scenario json/xml and queries test that were failing intermittently.
Another thing to keep in mind is careful usage of {force: true}. The action is performed if the element is available in the DOM and doesnt wait for animations and this could result in unnecessary failures. Tests ran ok 4 times successively. Hopefully it does so in Jenkins too.

Additionally pls review if this move to call toggleEntityConfirmation(false) within onNoClick wont impact the places where setCurrentQueryOnEntityChange() gets called.

Reply:
Verified scenarios for adding the toggleEntityConfirmation(false). Looks good to me, e2e tests passed too.